pkgs/applications/networking/instant-messengers/discord-screenaudio/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, mkDerivation
+, fetchFromGitHub
+, cmake
+, wrapQtAppsHook
+, pkg-config
+, qtbase
+, qtwebengine
+, knotifications
+, kxmlgui
+, kglobalaccel
+, pipewire
+, xdg-desktop-portal
+}:
+
+mkDerivation rec {
+  pname = "discord-screenaudio";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "maltejur";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-PPP/+7x0dcQHowB7hUZu85LK/G+ohrPeRB0vv6e3PBg=";
+    fetchSubmodules = true;
+  };
+
+  NIX_CFLAGS_COMPILE = [
+    "-I${pipewire.dev}/include/pipewire-0.3"
+    "-I${pipewire.dev}/include/spa-0.2"
+    "-Wno-pedantic" # Fails without flag
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    qtbase
+    qtwebengine
+    knotifications
+    kxmlgui
+    kglobalaccel
+    pipewire
+    pipewire.pulse
+    xdg-desktop-portal
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/maltejur/discord-screenaudio";
+    description = "A custom discord client that supports streaming with audio on Linux.";
+    license = licenses.gpl3;
+    mainProgram = "discord-screenaudio";
```

Comment:
I don't think this is technically necessary since without it, `nix run` will fallback to the derivation's name, but also it shouldn't break anything either.